src/producer/DefaultMQProducerImpl.cpp

```diff
@@ -410,7 +410,7 @@ SendResult DefaultMQProducerImpl::sendDefaultImpl(MQMessage& msg,
           case ComMode_ONEWAY:
             return sendResult;
           case ComMode_SYNC:
-            if (sendResult.getSendStatus() != SEND_OK) {
+            if (sendResult.getSendStatus() != SEND_OK && m_retryAnotherBrokerWhenNotStoreOk) {
```

Comment:
The default behavior is changed, should add docs

Reply:
Maybe. In Java it is `false`, and I keep it consistent.

https://github.com/apache/rocketmq/blob/26fa0501482bbf31c2a64a33f329ab9744ac3800/client/src/main/java/org/apache/rocketmq/client/producer/DefaultMQProducer.java#L124-L127